src/opnsense/mvc/app/models/OPNsense/Routing/Migrations/M1_0_0.php

```diff
@@ -0,0 +1,89 @@
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Routing\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Config;
+use OPNsense\Routing\Gateways;
+
+class M1_0_0 extends BaseModelMigration
+{
+    /**
+     * Migrate older models into shared model
+     * @param $model
+     */
+    public function run($model)
+    {
+        $config = Config::getInstance()->object();
+
+        if (!empty($config->gateways) && !empty($config->gateways->gateway_item)) {
+            foreach ($config->gateways->gateway_item as $gateway) {
+                $node = $model->gateway_item->Add();
+                $node_properties = iterator_to_array($node->iterateItems());
+
+                // apply non-existing/empty nodes
+                foreach ($node_properties as $key => $value) {
+                    if (empty((string)$gateway->$key)) {
+                        // either key doesn't exist or value is null
+                        $node->$key = (string)$value;
+                    }
+                }
+
+                // monitoring was on when no node present, exception to the rule above
+                $node->monitor_disable = !empty((string)$gateway->monitor_disable) ? '1' : '0';
+
+                // migrate set nodes
+                foreach ($gateway as $key => $value) {
+                    if (!array_key_exists($key, $node_properties)) {
+                        // skip unknown nodes
+                        continue;
+                    }
+
+                    if ($key === 'gateway') {
+                        // change all occurences of "dynamic" to empty string
+                        $node->gateway = str_replace('dynamic', '', (string)$value);
+                        continue;
+                    }
+
+                    $node->$key = (string)$value;
+                }
+
+                // apply dpinger defaults if old model didn't have them set
+                foreach ((new Gateways)::getDpingerDefaults() as $key => $value) {
```

Comment:
does this have to be constructed to be used? The function is declared as static.

Reply:
You're right, oversight on my end - at some point during development it was a regular public function